src/AzureCloudKeyVault.Core/KeyVaultManager.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.Management.KeyVault;
+using Microsoft.Azure.Management.KeyVault.Models;
+using Microsoft.Rest;
+using Squadron.AzureCloud;
+
+namespace Squadron
+{
+    public class KeyVaultManager
+
+    {
+        IKeyVaultManagementClient _client = null;
+        private readonly AzureCredentials _azureCredentials;
+        private readonly AzureResourceIdentifier _identifier;
+
+        internal KeyVaultManager(AzureCredentials azureCredentials,
+                                 AzureResourceIdentifier identifier)
+        {
+            _azureCredentials =
+                azureCredentials ?? throw new ArgumentNullException(nameof(azureCredentials));
+            _identifier =
+                identifier ?? throw new ArgumentNullException(nameof(identifier));
+        }
+
+        private async Task EnsureAuthenticatedAsync()
+        {
+            if (_client is null)
+            {
+                var tm = new AzureAdTokenManager();
+                TokenCredentials token = await tm.RequestTokenAsync(_azureCredentials);
+                _client = new KeyVaultManagementClient(token)
+                {
+                    SubscriptionId = _identifier.SubscriptionId,
+                };
+            }
+        }
+
+        public async Task<string> CreateAsync(string location)
+        {
+            await EnsureAuthenticatedAsync();
+            var name = $"squadron-{Guid.NewGuid().ToString("N").Substring(0, 8)}";
```

Comment:
this construct is repeated, maybe encase it in a single method to improve modifiability? just a suggestion as it is just a "one-liner"..